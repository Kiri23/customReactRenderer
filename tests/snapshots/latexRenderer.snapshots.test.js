const React = require("react");

// Import existing components
const {
  TikZDiagram,
  TikZCircle,
  TikZRectangle,
  TikZLine,
  TikZNode,
} = require("../../components/TikZComponents");
const {
  Document,
  Section,
  Paragraph,
} = require("../../components/LatexComponents");

// Import the existing latexRenderer
const latexRenderer = require("../../latexRenderer");

describe("latexRenderer - Snapshot Tests", () => {
  describe("Basic Document Snapshots", () => {
    test("should generate consistent basic document", () => {
      const BasicDocument = () => (
        <Document>
          <Section>Introduction</Section>
          <Paragraph>This is a basic test document.</Paragraph>
          <Section>Conclusion</Section>
          <Paragraph>This concludes the basic test.</Paragraph>
        </Document>
      );

      const output = latexRenderer.renderToLatex(<BasicDocument />);
      expect(output).toMatchSnapshot();
    });

    test("should generate consistent document with math", () => {
      const MathDocument = () => (
        <Document>
          <Section>Mathematical Content</Section>
          <Paragraph>
            The quadratic formula is: x = (-b ± √(b² - 4ac)) / 2a
          </Paragraph>
          <Paragraph>The area of a circle is: A = πr²</Paragraph>
        </Document>
      );

      const output = latexRenderer.renderToLatex(<MathDocument />);
      expect(output).toMatchSnapshot();
    });
  });

  describe("TikZ Component Snapshots", () => {
    test("should generate consistent TikZ diagram", () => {
      const TikZDocument = () => (
        <Document>
          <Section>TikZ Diagram</Section>
          <TikZDiagram width="8cm" height="6cm">
            <TikZCircle
              x={2}
              y={2}
              radius={1}
              options="fill=blue!20, draw=blue"
            />
            <TikZRectangle
              x={4}
              y={4}
              width={1.5}
              height={1}
              options="fill=red!20, draw=red"
            />
            <TikZLine from={[1, 1]} to={[5, 5]} options="thick, green" />
            <TikZNode x={2} y={3.5} text="Circle" options="above" />
            <TikZNode x={4.75} y={4.5} text="Rectangle" options="above" />
          </TikZDiagram>
        </Document>
      );

      const output = latexRenderer.renderToLatex(<TikZDocument />);
      expect(output).toMatchSnapshot();
    });

    test("should generate consistent complex TikZ diagram", () => {
      const ComplexTikZDocument = () => (
        <Document>
          <Section>Complex TikZ Diagram</Section>
          <TikZDiagram width="10cm" height="8cm">
            <TikZCircle x={1} y={1} radius={0.5} options="fill=yellow!30" />
            <TikZCircle x={3} y={3} radius={0.5} options="fill=green!30" />
            <TikZCircle x={5} y={5} radius={0.5} options="fill=purple!30" />
            <TikZLine from={[1, 1]} to={[3, 3]} options="dashed, thick" />
            <TikZLine from={[3, 3]} to={[5, 5]} options="dotted, thick" />
            <TikZRectangle
              x={0.5}
              y={0.5}
              width={1}
              height={1}
              options="fill=red!20"
            />
            <TikZNode x={2} y={2} text="Connection" options="above" />
            <TikZNode x={4} y={4} text="Connection" options="below" />
          </TikZDiagram>
        </Document>
      );

      const output = latexRenderer.renderToLatex(<ComplexTikZDocument />);
      expect(output).toMatchSnapshot();
    });

    test("should generate consistent TikZ with specific props", () => {
      const SpecificPropsDocument = () => (
        <Document>
          <Section>Specific Props Test</Section>
          <TikZDiagram width="6cm" height="4cm">
            <TikZCircle
              x={1.5}
              y={1.5}
              radius={0.8}
              options="fill=purple!40, draw=purple, thick"
            />
            <TikZRectangle
              x={3}
              y={2}
              width={2}
              height={1.5}
              options="fill=orange!30, draw=orange"
            />
            <TikZLine
              from={[0.5, 0.5]}
              to={[4.5, 3.5]}
              options="red, very thick"
            />
            <TikZNode
              x={1.5}
              y={2.3}
              text="Purple Circle"
              options="below, font=\\small"
            />
            <TikZNode
              x={4}
              y={2.75}
              text="Orange Rect"
              options="above, font=\\tiny"
            />
          </TikZDiagram>
        </Document>
      );

      const output = latexRenderer.renderToLatex(<SpecificPropsDocument />);
      expect(output).toMatchSnapshot();
    });
  });

  describe("Edge Cases Snapshots", () => {
    test("should generate consistent output for empty components", () => {
      const EmptyDocument = () => (
        <Document>
          <Section></Section>
          <Paragraph></Paragraph>
          <TikZDiagram>
            <TikZCircle x={0} y={0} radius={0} />
            <TikZRectangle x={0} y={0} width={0} height={0} />
            <TikZNode x={0} y={0} text="" />
          </TikZDiagram>
        </Document>
      );

      const output = latexRenderer.renderToLatex(<EmptyDocument />);
      expect(output).toMatchSnapshot();
    });

    test("should generate consistent output for extreme values", () => {
      const ExtremeValuesDocument = () => (
        <Document>
          <Section>Extreme Values</Section>
          <TikZDiagram width="1cm" height="1cm">
            <TikZCircle
              x={-1000}
              y={-1000}
              radius={1000}
              options="fill=black"
            />
            <TikZRectangle
              x={1000}
              y={1000}
              width={1000}
              height={1000}
              options="fill=white"
            />
            <TikZLine from={[-1000, -1000]} to={[1000, 1000]} options="thick" />
            <TikZNode x={0} y={0} text="Center" options="above" />
          </TikZDiagram>
        </Document>
      );

      const output = latexRenderer.renderToLatex(<ExtremeValuesDocument />);
      expect(output).toMatchSnapshot();
    });
  });

  describe("Nested Structure Snapshots", () => {
    test("should generate consistent deeply nested structure", () => {
      const DeepNestedDocument = () => (
        <Document>
          <Section>
            Level 1
            <Section>
              Level 2
              <Section>
                Level 3<Paragraph>Deep content</Paragraph>
                <TikZDiagram>
                  <TikZCircle x={1} y={1} radius={1} />
                  <TikZDiagram>
                    <TikZCircle x={2} y={2} radius={0.5} />
                  </TikZDiagram>
                </TikZDiagram>
              </Section>
            </Section>
          </Section>
        </Document>
      );

      const output = latexRenderer.renderToLatex(<DeepNestedDocument />);
      expect(output).toMatchSnapshot();
    });

    test("should generate consistent mixed content structure", () => {
      const MixedContentDocument = () => (
        <Document>
          <Section>Mixed Content Section</Section>
          <Paragraph>Regular paragraph text.</Paragraph>
          <TikZDiagram width="8cm" height="6cm">
            <TikZCircle x={2} y={2} radius={1} options="fill=blue!20" />
            <TikZNode x={2} y={3.5} text="Circle Label" options="above" />
          </TikZDiagram>
          <Paragraph>Another paragraph after diagram.</Paragraph>
          <Section>Another Section</Section>
          <Paragraph>Final paragraph.</Paragraph>
        </Document>
      );

      const output = latexRenderer.renderToLatex(<MixedContentDocument />);
      expect(output).toMatchSnapshot();
    });
  });

  describe("Multiple Format Snapshots", () => {
    test("should generate consistent enhanced LaTeX output", () => {
      const TestDocument = () => (
        <Document>
          <Section>Enhanced LaTeX Snapshot Test</Section>
          <Paragraph>This is an enhanced LaTeX snapshot test.</Paragraph>
          <TikZDiagram width="5cm" height="5cm">
            <TikZCircle x={2} y={2} radius={1} options="fill=green!20" />
          </TikZDiagram>
        </Document>
      );

      const output = latexRenderer.renderToEnhancedLatex(<TestDocument />);
      expect(output).toMatchSnapshot();
    });
  });

  describe("Performance Snapshot Tests", () => {
    test("should generate consistent large document", () => {
      const LargeDocument = () => {
        const sections = [];
        for (let i = 1; i <= 5; i++) {
          sections.push(<Section key={i}>Section {i}</Section>);
        }

        const tikzElements = [];
        for (let i = 0; i < 10; i++) {
          tikzElements.push(
            <TikZCircle
              key={i}
              x={i % 3}
              y={Math.floor(i / 3)}
              radius={0.2}
              options="fill=blue!20"
            />,
          );
        }

        return (
          <Document>
            {sections}
            <TikZDiagram width="6cm" height="6cm">
              {tikzElements}
            </TikZDiagram>
          </Document>
        );
      };

      const output = latexRenderer.renderToLatex(<LargeDocument />);
      expect(output).toMatchSnapshot();
    });
  });
});
