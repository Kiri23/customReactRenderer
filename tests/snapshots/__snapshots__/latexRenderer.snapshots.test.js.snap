// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`latexRenderer - Snapshot Tests Basic Document Snapshots should generate consistent basic document 1`] = `
"
\\documentclass{article}
\\usepackage{amsmath}
\\usepackage{amsfonts}
\\usepackage{amssymb}
\\usepackage{graphicx}
\\usepackage{tikz}
\\usepackage{pgfplots}
\\pgfplotsset{compat=1.18}
\\begin{document}


\\section{Introduction}
Introduction
This is a basic test document.


\\section{Conclusion}
Conclusion
This concludes the basic test.



\\end{document}"
`;

exports[`latexRenderer - Snapshot Tests Basic Document Snapshots should generate consistent document with math 1`] = `
"
\\documentclass{article}
\\usepackage{amsmath}
\\usepackage{amsfonts}
\\usepackage{amssymb}
\\usepackage{graphicx}
\\usepackage{tikz}
\\usepackage{pgfplots}
\\pgfplotsset{compat=1.18}
\\begin{document}


\\section{Mathematical Content}
Mathematical Content
The quadratic formula is: x = (-b ± √(b² - 4ac)) / 2a


The area of a circle is: A = πr²



\\end{document}"
`;

exports[`latexRenderer - Snapshot Tests Edge Cases Snapshots should generate consistent output for empty components 1`] = `
"
\\documentclass{article}
\\usepackage{amsmath}
\\usepackage{amsfonts}
\\usepackage{amssymb}
\\usepackage{graphicx}
\\usepackage{tikz}
\\usepackage{pgfplots}
\\pgfplotsset{compat=1.18}
\\begin{document}


\\section{}




\\begin{figure}[h]
\\centering
\\begin{tikzpicture}[scale=1]
\\draw (0,0) circle (0cm);
\\draw (0,0) rectangle (0,0);
\\node at (0,0) {};

\\end{tikzpicture}
\\end{figure}



\\end{document}"
`;

exports[`latexRenderer - Snapshot Tests Edge Cases Snapshots should generate consistent output for extreme values 1`] = `
"
\\documentclass{article}
\\usepackage{amsmath}
\\usepackage{amsfonts}
\\usepackage{amssymb}
\\usepackage{graphicx}
\\usepackage{tikz}
\\usepackage{pgfplots}
\\pgfplotsset{compat=1.18}
\\begin{document}


\\section{Extreme Values}
Extreme Values
\\begin{figure}[h]
\\centering
\\begin{tikzpicture}[scale=1]
\\draw[fill=black] (-1000,-1000) circle (1000cm);
\\draw[fill=white] (1000,1000) rectangle (2000,2000);
\\draw[thick] (-1000,-1000) -- (1000,1000);
\\node[above] at (0,0) {Center};

\\end{tikzpicture}
\\end{figure}



\\end{document}"
`;

exports[`latexRenderer - Snapshot Tests Multiple Format Snapshots should generate consistent enhanced LaTeX output 1`] = `
"
\\documentclass{article}
\\usepackage{amsmath}
\\usepackage{amsfonts}
\\usepackage{amssymb}
\\usepackage{graphicx}
\\usepackage{tikz}
\\usepackage{pgfplots}
\\pgfplotsset{compat=1.18}
\\begin{document}


\\section{Enhanced LaTeX Snapshot Test}
Enhanced LaTeX Snapshot Test
This is an enhanced LaTeX snapshot test.


\\begin{figure}[h]
\\centering
\\begin{tikzpicture}[scale=1]
\\draw[fill=green!20] (2,2) circle (1cm);

\\end{tikzpicture}
\\end{figure}



\\end{document}"
`;

exports[`latexRenderer - Snapshot Tests Nested Structure Snapshots should generate consistent deeply nested structure 1`] = `
"
\\documentclass{article}
\\usepackage{amsmath}
\\usepackage{amsfonts}
\\usepackage{amssymb}
\\usepackage{graphicx}
\\usepackage{tikz}
\\usepackage{pgfplots}
\\pgfplotsset{compat=1.18}
\\begin{document}


\\section{Level 1
\\section{Level 2
\\section{Level 3
Deep content


\\begin{figure}[h]
\\centering
\\begin{tikzpicture}[scale=1]
\\draw (1,1) circle (1cm);

\\begin{figure}[h]
\\centering
\\begin{tikzpicture}[scale=1]
\\draw (2,2) circle (0.5cm);

\\end{tikzpicture}
\\end{figure}


\\end{tikzpicture}
\\end{figure}

}
Level 3
Deep content


\\begin{figure}[h]
\\centering
\\begin{tikzpicture}[scale=1]
\\draw (1,1) circle (1cm);

\\begin{figure}[h]
\\centering
\\begin{tikzpicture}[scale=1]
\\draw (2,2) circle (0.5cm);

\\end{tikzpicture}
\\end{figure}


\\end{tikzpicture}
\\end{figure}

}
Level 2
\\section{Level 3
Deep content


\\begin{figure}[h]
\\centering
\\begin{tikzpicture}[scale=1]
\\draw (1,1) circle (1cm);

\\begin{figure}[h]
\\centering
\\begin{tikzpicture}[scale=1]
\\draw (2,2) circle (0.5cm);

\\end{tikzpicture}
\\end{figure}


\\end{tikzpicture}
\\end{figure}

}
Level 3
Deep content


\\begin{figure}[h]
\\centering
\\begin{tikzpicture}[scale=1]
\\draw (1,1) circle (1cm);

\\begin{figure}[h]
\\centering
\\begin{tikzpicture}[scale=1]
\\draw (2,2) circle (0.5cm);

\\end{tikzpicture}
\\end{figure}


\\end{tikzpicture}
\\end{figure}

}
Level 1
\\section{Level 2
\\section{Level 3
Deep content


\\begin{figure}[h]
\\centering
\\begin{tikzpicture}[scale=1]
\\draw (1,1) circle (1cm);

\\begin{figure}[h]
\\centering
\\begin{tikzpicture}[scale=1]
\\draw (2,2) circle (0.5cm);

\\end{tikzpicture}
\\end{figure}


\\end{tikzpicture}
\\end{figure}

}
Level 3
Deep content


\\begin{figure}[h]
\\centering
\\begin{tikzpicture}[scale=1]
\\draw (1,1) circle (1cm);

\\begin{figure}[h]
\\centering
\\begin{tikzpicture}[scale=1]
\\draw (2,2) circle (0.5cm);

\\end{tikzpicture}
\\end{figure}


\\end{tikzpicture}
\\end{figure}

}
Level 2
\\section{Level 3
Deep content


\\begin{figure}[h]
\\centering
\\begin{tikzpicture}[scale=1]
\\draw (1,1) circle (1cm);

\\begin{figure}[h]
\\centering
\\begin{tikzpicture}[scale=1]
\\draw (2,2) circle (0.5cm);

\\end{tikzpicture}
\\end{figure}


\\end{tikzpicture}
\\end{figure}

}
Level 3
Deep content


\\begin{figure}[h]
\\centering
\\begin{tikzpicture}[scale=1]
\\draw (1,1) circle (1cm);

\\begin{figure}[h]
\\centering
\\begin{tikzpicture}[scale=1]
\\draw (2,2) circle (0.5cm);

\\end{tikzpicture}
\\end{figure}


\\end{tikzpicture}
\\end{figure}



\\end{document}"
`;

exports[`latexRenderer - Snapshot Tests Nested Structure Snapshots should generate consistent mixed content structure 1`] = `
"
\\documentclass{article}
\\usepackage{amsmath}
\\usepackage{amsfonts}
\\usepackage{amssymb}
\\usepackage{graphicx}
\\usepackage{tikz}
\\usepackage{pgfplots}
\\pgfplotsset{compat=1.18}
\\begin{document}


\\section{Mixed Content Section}
Mixed Content Section
Regular paragraph text.


\\begin{figure}[h]
\\centering
\\begin{tikzpicture}[scale=1]
\\draw[fill=blue!20] (2,2) circle (1cm);
\\node[above] at (2,3.5) {Circle Label};

\\end{tikzpicture}
\\end{figure}


Another paragraph after diagram.


\\section{Another Section}
Another Section
Final paragraph.



\\end{document}"
`;

exports[`latexRenderer - Snapshot Tests Performance Snapshot Tests should generate consistent large document 1`] = `
"
\\documentclass{article}
\\usepackage{amsmath}
\\usepackage{amsfonts}
\\usepackage{amssymb}
\\usepackage{graphicx}
\\usepackage{tikz}
\\usepackage{pgfplots}
\\pgfplotsset{compat=1.18}
\\begin{document}


\\section{Section 1}
Section 1
\\section{Section 2}
Section 2
\\section{Section 3}
Section 3
\\section{Section 4}
Section 4
\\section{Section 5}
Section 5
\\begin{figure}[h]
\\centering
\\begin{tikzpicture}[scale=1]
\\draw[fill=blue!20] (0,0) circle (0.2cm);
\\draw[fill=blue!20] (1,0) circle (0.2cm);
\\draw[fill=blue!20] (2,0) circle (0.2cm);
\\draw[fill=blue!20] (0,1) circle (0.2cm);
\\draw[fill=blue!20] (1,1) circle (0.2cm);
\\draw[fill=blue!20] (2,1) circle (0.2cm);
\\draw[fill=blue!20] (0,2) circle (0.2cm);
\\draw[fill=blue!20] (1,2) circle (0.2cm);
\\draw[fill=blue!20] (2,2) circle (0.2cm);
\\draw[fill=blue!20] (0,3) circle (0.2cm);

\\end{tikzpicture}
\\end{figure}



\\end{document}"
`;

exports[`latexRenderer - Snapshot Tests TikZ Component Snapshots should generate consistent TikZ diagram 1`] = `
"
\\documentclass{article}
\\usepackage{amsmath}
\\usepackage{amsfonts}
\\usepackage{amssymb}
\\usepackage{graphicx}
\\usepackage{tikz}
\\usepackage{pgfplots}
\\pgfplotsset{compat=1.18}
\\begin{document}


\\section{TikZ Diagram}
TikZ Diagram
\\begin{figure}[h]
\\centering
\\begin{tikzpicture}[scale=1]
\\draw[fill=blue!20, draw=blue] (2,2) circle (1cm);
\\draw[fill=red!20, draw=red] (4,4) rectangle (5.5,5);
\\draw[thick, green] (1,1) -- (5,5);
\\node[above] at (2,3.5) {Circle};
\\node[above] at (4.75,4.5) {Rectangle};

\\end{tikzpicture}
\\end{figure}



\\end{document}"
`;

exports[`latexRenderer - Snapshot Tests TikZ Component Snapshots should generate consistent TikZ with specific props 1`] = `
"
\\documentclass{article}
\\usepackage{amsmath}
\\usepackage{amsfonts}
\\usepackage{amssymb}
\\usepackage{graphicx}
\\usepackage{tikz}
\\usepackage{pgfplots}
\\pgfplotsset{compat=1.18}
\\begin{document}


\\section{Specific Props Test}
Specific Props Test
\\begin{figure}[h]
\\centering
\\begin{tikzpicture}[scale=1]
\\draw[fill=purple!40, draw=purple, thick] (1.5,1.5) circle (0.8cm);
\\draw[fill=orange!30, draw=orange] (3,2) rectangle (5,3.5);
\\draw[red, very thick] (0.5,0.5) -- (4.5,3.5);
\\node[below, font=\\\\small] at (1.5,2.3) {Purple Circle};
\\node[above, font=\\\\tiny] at (4,2.75) {Orange Rect};

\\end{tikzpicture}
\\end{figure}



\\end{document}"
`;

exports[`latexRenderer - Snapshot Tests TikZ Component Snapshots should generate consistent complex TikZ diagram 1`] = `
"
\\documentclass{article}
\\usepackage{amsmath}
\\usepackage{amsfonts}
\\usepackage{amssymb}
\\usepackage{graphicx}
\\usepackage{tikz}
\\usepackage{pgfplots}
\\pgfplotsset{compat=1.18}
\\begin{document}


\\section{Complex TikZ Diagram}
Complex TikZ Diagram
\\begin{figure}[h]
\\centering
\\begin{tikzpicture}[scale=1]
\\draw[fill=yellow!30] (1,1) circle (0.5cm);
\\draw[fill=green!30] (3,3) circle (0.5cm);
\\draw[fill=purple!30] (5,5) circle (0.5cm);
\\draw[dashed, thick] (1,1) -- (3,3);
\\draw[dotted, thick] (3,3) -- (5,5);
\\draw[fill=red!20] (0.5,0.5) rectangle (1.5,1.5);
\\node[above] at (2,2) {Connection};
\\node[below] at (4,4) {Connection};

\\end{tikzpicture}
\\end{figure}



\\end{document}"
`;
